pkg/endpoint/policy.go

```diff
@@ -696,6 +705,10 @@ func (e *Endpoint) regenerate(owner Owner, reason string) (retErr error) {
 	e.bumpPolicyRevision(revision)
 
 	e.getLogger().Info("Endpoint policy recalculated")
+	regenerateEnd := time.Since(regenerateStart) / time.Second
```

Comment:
You should put this code into a `defer` statement, in order to not ignore regenerations that fail.

Reply:
The reason I added this at the very end was because I was not sure if I need to take a lock for `e.getLogger()` in the defer function.  At the end of the function, when we are returning success, and there was already a `e.getLogger()` being called without locking, I thought of adding this after that. I guess I was simply trying to avoid taking a lock if possible. What do you think?

Reply:
If we need to use lock `e.Mutex.RLock()` for e.getLogger(), I am wondering if the existing code with `e.getLogger().Info("Endpoint policy recalculated")` was incorrect and also needs locking.

Reply:
done